common/reconciliation/fetcher/timer_test.go

```diff
@@ -0,0 +1,177 @@
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package fetcher
+
+import (
+	"context"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+	"github.com/uber/cadence/common/pagination"
+	"github.com/uber/cadence/common/persistence"
+	"github.com/uber/cadence/common/reconciliation/entity"
+)
+
+func TestGetUserTimers(t *testing.T) {
+	fixedTimestamp, err := time.Parse(time.RFC3339, "2023-12-12T22:08:41Z")
+	if err != nil {
+		t.Fatalf("Failed to parse timestamp: %v", err)
+	}
+
+	ctrl := gomock.NewController(t)
+
+	mockRetryer := persistence.NewMockRetryer(ctrl)
+	ctx := context.Background()
+	pageSize := 10
+	minTimestamp := fixedTimestamp.Add(-time.Hour)
+	maxTimestamp := fixedTimestamp
+	nonNilToken := []byte("non-nil-token")
+	testCases := []struct {
+		name          string
+		setupMock     func() []*persistence.TimerTaskInfo
+		expectedPage  pagination.Page
+		expectedError bool
+	}{
+		{
+			name: "Success",
+			setupMock: func() []*persistence.TimerTaskInfo {
+				timerTasks := []*persistence.TimerTaskInfo{
+					{
+						DomainID:            "testDomainID",
+						WorkflowID:          "testWorkflowID",
+						RunID:               "testRunID",
+						VisibilityTimestamp: fixedTimestamp,
+						TaskType:            persistence.TaskTypeUserTimer,
+					},
+				}
+
+				mockRetryer.EXPECT().
+					GetTimerIndexTasks(ctx, &persistence.GetTimerIndexTasksRequest{
+						MinTimestamp:  minTimestamp,
+						MaxTimestamp:  maxTimestamp,
+						BatchSize:     pageSize,
+						NextPageToken: nil,
+					}).
+					Return(&persistence.GetTimerIndexTasksResponse{
+						Timers:        timerTasks,
+						NextPageToken: nil,
+					}, nil)
+
+				mockRetryer.EXPECT().GetShardID().Return(123)
+
+				return timerTasks
+			},
+			expectedPage: pagination.Page{
+				Entities: []pagination.Entity{
+					&entity.Timer{
+						ShardID:             123,
+						DomainID:            "testDomainID",
+						WorkflowID:          "testWorkflowID",
+						RunID:               "testRunID",
+						TaskType:            persistence.TaskTypeUserTimer,
+						VisibilityTimestamp: fixedTimestamp,
+					},
+				},
+			},
+			expectedError: false,
+		},
+		{
+			name: "Non-nil Pagination Token Passed",
```

Comment:
nit: a bit confusing to see Passed in a test name. Will be especially confusing to see "Something ... Passed Failed" in IDE :-) Consider renaming to "Provided".